Task 4: Submission

redshadowhero opened the pull request:
This is my submission.

A few notes:
* The Docker build folder, systemd service file, docker repo file (for yum), and the bash script that echos the date are in `playbooks/roles/docker/files`
* This has been tested 'end to end' on a fresh Centos 7 install; most of the documentation was written from this test. It's also been tested on Mac OS X Sierra and Debian 7, but those machines have already been set up for development, so I'm not 100% sure if any extra development files are needed other than the ones noted in `INSTALL.md`

Please let me know if you have any questions!

kmullican commented on INSTALL.md:
```diff
@@ -0,0 +1,84 @@
+3. virtualenv
+4. gcc
+5. rsync
+
```
dos2unix

kmullican commented on INSTALL.md:
````diff
@@ -0,0 +1,84 @@
+```
+
+Also, git mangles the permission on the ssh keys, correct them with: `chmod 0600 misc/hudsonalpha*`
+
````
Be sure all files are in unix format by running the following command (not doing so causes errors when executing do/setup.py, among other files)

find /path/to/code -type f -exec dos2unix {} \;

kmullican commented on playbooks/roles/docker/files/image/Dockerfile:
```diff
@@ -0,0 +1,8 @@
+MAINTAINER "Chris King" <tchrisking@gmail.com>
+
+COPY datetime /usr/local/bin/datetime
+
```
RUN chmod +x /usr/local/bin/datetime

kmullican commented on playbooks/roles/docker/files/datetime.service:
```diff
@@ -0,0 +1,14 @@
+ExecStartPre=-/usr/bin/docker stop datetime
+ExecStartPre=-/usr/bin/docker rm datetime
+ExecStart=/usr/bin/docker run -d --name datetime datetime:habuild
+ExecStopPre=-/usr/bin/docker stop -t 11 datetime
```
ExecStopPre causes error when running this service -- ExecStop (without the "Pre") runs OK

redshadowhero commented:
I fixed the noted issues in 092c7daac960ed4b66f7daf9a60a697f7a3180cf

Nice catch on the service file -- sometimes I'll yank a line in vim and paste it elsewhere, changing one thing while leaving stuff tacked in that I really shouldn't have.

I also fired up a Fedora 25 (Workstation edition, but I don't think it matters) for this test, but I didn't encounter the dos encoding issue. Still, it's probably better to be verbose, so I updated the corresponding documentation.
